tests/server.rs

```diff
@@ -2743,6 +2743,31 @@ fn http1_trailer_fields_not_allowed() {
     assert_eq!(body, expected_body);
 }
 
+#[test]
+fn http1_trailer_recv_fields() {
+    let server = serve();
+    let mut req = connect(server.addr());
+    req.write_all(
+        b"\
+        POST / HTTP/1.1\r\n\
+        trailer: chunky-trailer\r\n\
+        host: example.domain\r\n\
+        transfer-encoding: chunked\r\n\
+        \r\n\
+        5\r\n\
+        hello\r\n\
+        0\r\n\
+        chunky-trailer: header data\r\n\
+        \r\n\
+    ",
+    )
+    .expect("writing");
+
+    assert_eq!(server.body(), b"hello");
+
+    // FIXME: add support for checking trailers that server received
```

Comment:
I will add this tomorrow.

Reply:
added in [1d5eaf4](https://github.com/hyperium/hyper/pull/3637/commits/1d5eaf4979628d51a274c560bbab88e21864b3ed)